ui/panels-plugin/src/plugins/time-series-chart/VisualOptionsEditor.tsx

```diff
@@ -105,6 +105,27 @@ export function VisualOptionsEditor({ value, onChange }: VisualOptionsEditorProp
           />
         }
       />
+      <OptionsEditorControl
+        label={'Palette'}
+        control={
+          <ToggleButtonGroup
+            color="primary"
+            exclusive
+            value={value.palette?.kind ?? 'Auto'}
+            onChange={(__, newValue) => {
+              const palette: VisualOptions['palette'] =
```

Comment:
Optional nit: it might be nice to pull this handler logic into a function. I find it's a little easier to read/maintain.

Reply:
This will be deleted in follow-up PR #1124 based on feedback from Mohit